packages/@apphosting/adapter-angular/src/bin/build.ts

```diff
@@ -1,20 +1,87 @@
 #! /usr/bin/env node
 import { spawn } from "child_process";
-import { loadConfig } from "../utils.js";
+import { loadConfig, ApplicationBuilderOptions, move, writeFile } from "../utils.js";
+import { resolve, normalize, relative } from "path";
+import { stringify as yamlStringify } from "yaml";
 
+interface OutputBundleOptions {
+  bundleYamlPath: string;
+  outputDirectory: string;
+  outputBaseDirectory: string;
+  serverFilePath: string;
+  browserDirectory: string;
+}
+
+const cwd = process.cwd();
+
+// read Application Builder config
+const config: ApplicationBuilderOptions = await loadConfig(cwd);
+const outputPath = config.outputPath;
+// normalized output path structure
+const normalizedOutputPath = {
+  browser: "browser",
+  server: "server",
+  media: "media",
+  ...(typeof outputPath === "string" ? undefined : outputPath),
+  base: normalize(resolve(typeof outputPath === "string" ? outputPath : outputPath.base)),
+};
+
+// populate file/directory paths we need inside app hosting output directory
+function populateOutputBundleOptions(): OutputBundleOptions {
+  const outputBundleDir = resolve(".apphosting");
+  return {
+    bundleYamlPath: resolve(outputBundleDir, "bundle.yaml"),
+    outputDirectory: outputBundleDir,
+    outputBaseDirectory: resolve(outputBundleDir, "dist"),
+    serverFilePath: resolve(outputBundleDir, "dist", normalizedOutputPath.server, "server.mjs"),
+    browserDirectory: resolve(outputBundleDir, "dist", normalizedOutputPath.browser),
+  };
+}
+
+// Run build command
 const build = (cwd = process.cwd()) =>
   new Promise<void>((resolve, reject) => {
-    // TODO warn if the build script contains anything other than `ng build`
     const process = spawn("npm", ["run", "build"], { cwd, shell: true, stdio: "inherit" });
     process.on("exit", (code) => {
       if (code === 0) return resolve();
       reject();
     });
   });
 
-const config = await loadConfig(process.cwd());
+// move the base output directory, which contains the server and browser bundle directory, and prerendered routes
+// as well as generating bundle.yaml
+async function generateOutputDirectory(
```

Comment:
Could we move this function + populateOutputBundleOptions + any other dependent functions into utils.js? It will help facilitate the unit tests

Reply:
done